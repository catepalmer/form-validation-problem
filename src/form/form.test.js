import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import 'jest-enzyme'
import React from 'react'

import Form from './form.component'

Enzyme.configure({ adapter: new Adapter() })

describe('components:Form', () => {
  it('renders the Form component', () => {
    expect(toJson(shallow(<Form />))).toMatchSnapshot()
  })
})
