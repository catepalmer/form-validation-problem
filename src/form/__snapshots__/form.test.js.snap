// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:Form renders the Form component 1`] = `
<form
  action=""
  method="post"
  onSubmit={[Function]}
>
  <h1>
    Fill out this awesome form
  </h1>
  <fieldset>
    <h3>
      Your details
    </h3>
    <p
      className=""
    >
      <label
        className="label"
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        name="email"
        type="text"
      />
    </p>
    <p
      className=""
    >
      <label
        className="label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        name="password"
        type="password"
      />
    </p>
  </fieldset>
  <fieldset>
    <h3>
      Your animal
    </h3>
    <p
      className=""
    >
      <label
        className="label"
        htmlFor="colour"
      >
        Colour
      </label>
      <select
        id="colour"
        name="colour"
      >
        <option
          value=""
        >
          Choose colour
        </option>
        <option
          value="blue"
        >
          Blue
        </option>
        <option
          value="green"
        >
          Green
        </option>
        <option
          value="red"
        >
          Red
        </option>
        <option
          value="black"
        >
          Black
        </option>
        <option
          value="brown"
        >
          Brown
        </option>
      </select>
    </p>
    <p
      className=""
    >
      <span
        className="label"
      >
        Animal
      </span>
      <label
        htmlFor="bear"
      >
        <input
          id="bear"
          name="animals"
          type="checkbox"
          value="bear"
        />
        Bear
      </label>
      <label
        htmlFor="tiger"
      >
        <input
          id="tiger"
          name="animals"
          type="checkbox"
          value="tiger"
        />
        Tiger
      </label>
      <label
        htmlFor="snake"
      >
        <input
          id="snake"
          name="animals"
          type="checkbox"
          value="snake"
        />
        Snake
      </label>
      <label
        htmlFor="donkey"
      >
        <input
          id="donkey"
          name="animals"
          type="checkbox"
          value="donkey"
        />
        Donkey
      </label>
    </p>
    <p
      className=""
    >
      <label
        className="label"
        htmlFor="tigerType"
      >
        Type of tiger
      </label>
      <input
        id="tigerType"
        name="tigerType"
        type="text"
      />
    </p>
  </fieldset>
  <fieldset>
    <p>
      <input
        type="submit"
        value="Create account"
      />
    </p>
  </fieldset>
</form>
`;
